src/npc_attack.cpp

```diff
@@ -384,30 +470,30 @@ bool npc_attack_throw::can_use( const npc &source ) const
     if( single_item.count_by_charges() ) {
         single_item.charges = 1;
     }
-    // please don't throw your pants...
-    return !source.is_worn( thrown_item ) &&
-           // we would rather throw than activate this. unless we need to throw it now.
-           !( thrown_item.has_flag( flag_NPC_THROW_NOW ) || thrown_item.has_flag( flag_NPC_ACTIVATE ) ) &&
-           // i don't trust you as far as i can throw you.
-           source.throw_range( single_item ) != 0;
+    if( thrown_item.has_flag( flag_NPC_THROW_NOW ) || thrown_item.has_flag( flag_NPC_THROWN ) ) {
+        //Always allow throwing items that are flagged as throw now or npc_thrown
+        return true;
+    }
+
+    bool throwable = source.throw_range( single_item ) > 0 && !source.is_worn( thrown_item ) &&
+                     !thrown_item.has_flag( flag_NPC_ACTIVATE );
```

Comment:
you reversed tihs. are you sure this is correct that you can't throw items that have NPC_ACTIVATE?

Reply:
Had to double check. NPC_ACTIVATE is the inactive version of explosives. When they are activated they loose the flag so throwing NPC_ACTIVATE items would mean throwing inactive grenades.